src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -54,55 +59,11 @@ public function hasCacheableSupportsMethod(): bool
      */
     protected function extractAttributes($object, $format = null, array $context = array())
     {
-        // If not using groups, detect manually
-        $attributes = array();
-
-        // methods
-        $reflClass = new \ReflectionClass($object);
-        foreach ($reflClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflMethod) {
-            if (
-                0 !== $reflMethod->getNumberOfRequiredParameters() ||
-                $reflMethod->isStatic() ||
-                $reflMethod->isConstructor() ||
-                $reflMethod->isDestructor()
-            ) {
-                continue;
-            }
-
-            $name = $reflMethod->name;
-            $attributeName = null;
-
-            if (0 === strpos($name, 'get') || 0 === strpos($name, 'has')) {
-                // getters and hassers
-                $attributeName = substr($name, 3);
-
-                if (!$reflClass->hasProperty($attributeName)) {
-                    $attributeName = lcfirst($attributeName);
-                }
-            } elseif (0 === strpos($name, 'is')) {
-                // issers
-                $attributeName = substr($name, 2);
-
-                if (!$reflClass->hasProperty($attributeName)) {
-                    $attributeName = lcfirst($attributeName);
-                }
-            }
-
-            if (null !== $attributeName && $this->isAllowedAttribute($object, $attributeName, $format, $context)) {
-                $attributes[$attributeName] = true;
-            }
-        }
-
-        // properties
-        foreach ($reflClass->getProperties(\ReflectionProperty::IS_PUBLIC) as $reflProperty) {
-            if ($reflProperty->isStatic() || !$this->isAllowedAttribute($object, $reflProperty->name, $format, $context)) {
-                continue;
-            }
-
-            $attributes[$reflProperty->name] = true;
-        }
+        $properties = $this->propertyListExtractor->getProperties(get_class($object), ['exclude_static_properties' => true]);
 
-        return array_keys($attributes);
+        return array_filter($properties, function (string $attribute) use ($object, $format, $context) {
```

Comment:
Using `foreach` will probably be faster than calling `array_filter` (because it triggers less function calls). As it's on the hot path, it is worth it.